### PR TITLE
Play video on click

### PR DESCRIPTION
**Pull request by wgranger**
Describe your changes.
This allows the ability for a video to play when clicking on the still image. A boolean prop is also added to give the component the ability to hide controls, if needed. Bat detective has video subjects to test functionality.

# Review Checklist

- [x] Does it work in all major browsers: Firefox, Chrome, Edge, Safari?
- [x] Does it work on mobile?
- [x] Can you `rm -rf node_modules/ && npm install` and app works as expected?
- [x] Did you deploy a staging branch?
- [ ] Did you get any type checking errors from [Babel Typecheck](https://github.com/codemix/babel-plugin-typecheck)

https://video-click-play.pfe-preview.zooniverse.org/

## Optional

- [ ] If it's a new component, is it in ES6? Is it clear of warnings from ESLint?
- [ ] Have you replaced any `ChangeListener` or `PromiseRenderer` components with code that updates component state?
- [ ] If changes are made to the classifier, does the dev classifier still work?
- [ ] Have you added in [flow type annotations](https://flowtype.org/docs/type-annotations.html)?

### CODE REVIEW

**Comment by wgranger** [comment]
The linter complained about adding a click listener to a `<video>` tag, so an invisible button overlay was added. I'm open to other suggestions or reverting back to the video click listener.

**Comment by eatyourgreens** [comment]
Clicks on the video element are being blocked by the SVG image layer. Adding another transparent button on top isn't the way to go to solve this, and will look like a clickjacking technique to any security plugins.

I'd suggest just remove the SVG layer so that the video element can be clicked directly.

File: app/components/video-player.jsx [hunk 1]
```diff
@@ -84,14 +91,19 @@ const VideoPlayer = React.createClass({
...
     return (
       <div className="subject-video-frame">
+        <button
+          className="subject-video-frame__hidden-button"
+          onClick={this.playVideo.bind(this, !this.state.playing)}
+        />
+
         <video
           className="subject"
           ref="videoPlayer"
```
Comment (eatyourgreens) [inline app/components/video-player.jsx]: I think you've missed out `controls={!this.props.showControls}` on the `video` element here. I'm assuming that the browser's native controls should show if the custom controls are turned off.

Comment (wgranger) [inline app/components/video-player.jsx]: The idea was to hide all controls for the video element, including native controls. This was to adhere to styling for the new project landing page.

**Comment by wgranger** [comment]
Ok, the `onClick` was added to the video element. This onClick actually worked before, with or without the svg layer, but the linter gave warnings about attaching an onClick to that element.

**Comment by wgranger** [comment]
screen shot 2016-12-19 at 1 57 36 pm

This is an example of the styling. However, if these subjects were videos, they could be played with a click.

**Comment by eatyourgreens** [comment]
I'm a bit confused, because you can already click the video element to play videos outside the classifier e.g.  https://www.zooniverse.org/projects/zooniverse/arizona-batwatch/talk/230/146363

**Comment by wgranger** [comment]
I think that link allows you to play videos on click on mobile, but not desktop.

**Comment by eatyourgreens** [comment]
How about subjects that use the `Thumbnail` component instead of the `SubjectViewer`? https://www.zooniverse.org/projects/zooniverse/arizona-batwatch/talk/tags/insect

**Comment by eatyourgreens** [comment]
Was the linter warning about adding `onClick` to an element that can't receive keyboard focus? That's pretty easy to fix too.

Otherwise, this looks good to me.

**Comment by eatyourgreens** [comment]
I've just done a quick check using the keyboard in Chrome for Arizona Batwatch and the native video controls at https://www.zooniverse.org/projects/zooniverse/arizona-batwatch/talk/tags/insect. You can tab to individual videos and the video element will receive focus. You can then tab to the play button, within the video element (it's hidden in the shadow DOM for that element), and toggle it with the spacebar when it has focus.

I didn't check any other browsers, but I think that `jsx-a11y` warning about adding click to elements without corresponding keyboard interaction is wrong here. Keyboard support might vary between browsers, though.

**Comment by wgranger** [comment]
No, the particular linter gave the warning, `visible, non-interactive elements should not have mouse or keyboard event listeners`

**Comment by eatyourgreens** [comment]
Yeah, that's an accessibility warning from `jsx-a11y`. Fine to ignore it here, if the video can be controlled with the keyboard.

**Pull request merged**